maliput/src/utility/mod.rs

```diff
@@ -53,18 +53,27 @@ pub type ObjFeatures = maliput_sys::utility::ffi::Features;
 pub fn generate_obj_file(
     road_network: &RoadNetwork,
     dirpath: impl AsRef<Path>,
-    fileroot: impl AsRef<Path>,
+    fileroot: &String,
     obj_features: &ObjFeatures,
 ) -> Result<PathBuf, Box<dyn Error>> {
-    let complete_file_path = dirpath.as_ref().join(fileroot.as_ref().with_extension("obj"));
+    // Saves the complete path to the generated Wavefront file.
+    let future_obj_file_path = dirpath.as_ref().join(fileroot.clone() + ".obj");
     let dirpath = to_string(dirpath)?;
-    let fileroot = to_string(fileroot)?;
+    // Creates dirpath if does not exist.
+    if !Path::new(&dirpath).exists() {
```

Comment:
Genuinely curious, does `create_all_dir` actually fail if the directory already exists?